sdk/identity/azure-identity/inc/azure/identity/client_certificate_credential.hpp

```diff
@@ -30,7 +30,7 @@ namespace Azure { namespace Identity {
     template <> struct UniquePrivateKeyHelper<void*>
     {
       static void FreePrivateKey(void* pkey) { FreePrivateKeyImpl(pkey); }
-      using type = Azure::Core::_internal::BasicUniqueHandle<void, FreePrivateKey>;
+      using type = Azure::Core::_detail::BasicUniqueHandle<void, FreePrivateKey>;
```

Comment:
If identity ends up using `_detail` namespace from another library, it is a problem. Either Identity needs to declare its own thing, or that thing from Core really needs to be `_internal`.

Reply:
The problem with this header being _internal is that it means that we pollute API views with unrelated types for any components which are used by other packages (Storage::Common, Azure::Core::Amqp, Azure::Core).

The entire purpose of this is to remove the underlying type from ApiView.

For Identity, this isn't a problem because `_internal` types aren't included in the ApiView. But for common packages, they are included in the ApiView since they have similar contractual obligations to public APIs.